app/controllers/openid_controller.rb

```diff
@@ -55,7 +55,7 @@ def index
     if current_user.nil? && params['openid.mode'] != 'check_authentication'
       session[:openid_return_to] = request.env['ORIGINAL_FULLPATH']
       flash[:warning] = 'Please log in first.'
-      redirect_to '/login'
+      redirect_to '/auth/github'
```

Comment:
I want here the line 
if provider != nil
..... then redirect_to '/auth/' + provider
else
......redirect_to '/login'
end
I am trying hard to get provider variable from the MK.

Reply:
One step closer. The route now has the provider. 
I cannot find the linking between the sequence of events as written earlier in https://github.com/publiclab/mapknitter/issues/378#issuecomment-485387514 and this controller action.

Reply:
Once we will achieve this then pr will be completed. So, we are on the last step. Cool. Great workflow here I did.